src/Lumper.UI/Views/MainWindow.axaml.cs

```diff
@@ -77,4 +83,19 @@ private void PageButton_OnClick(object? _, RoutedEventArgs e)
         if (source is not null && !source.IsChecked!.Value)
             source.IsChecked = true;
     }
+
+    private async void Window_OnDrop(object? _, DragEventArgs e)
+    {
+        List<IStorageFile> files = FilterDraggedBspFiles(e);
+        await ViewModel!.OpenDragDropped(files);
+    }
+
+    private void Window_OnDragOver(object? _, DragEventArgs e)
+    {
+        bool hasBsps = FilterDraggedBspFiles(e).Count > 0;
+        e.DragEffects = hasBsps ? DragDropEffects.Link : DragDropEffects.None;
+    }
+
+    private List<IStorageFile> FilterDraggedBspFiles(DragEventArgs e) =>
+        e.Data.GetFiles()?.OfType<IStorageFile>().Where(file => file.Name.EndsWith(".bsp")).ToList() ?? [];
```

Comment:
I think you need to add null propagation after `OfType<IStorageFile>()`:
`e.Data.GetFiles()?.OfType<IStorageFile>()?.Where(file => file.Name.EndsWith(".bsp")).ToList() ?? [];`

Reply:
In C#, unlike JS, `?.` short-circuits https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/member-access-operators#null-conditional-operators--and-

Also note that the nulluable reference type static analysis is very good at flagging this -- if you applied your suggestion the static analysis would raise a warning and release compiles would fail.